xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/metadata/DocumentSolrMetadataExtractor.java

```diff
@@ -69,6 +72,15 @@
     @Inject
     private EntityReferenceSerializer<String> entityReferenceSerializer;
 
+    @Inject
+    private UserReferenceSerializer<String> userReferenceSerializer;
+
+    // TODO: relying on this serializer should be prevented by properly rewriting XWiki#getPlainUserName
+    //  to use UserReference
+    @Inject
+    @Named("document")
+    private UserReferenceSerializer<DocumentReference> documentReferenceUserReferenceSerializer;
```

Comment:
Note that fixing that won't help reducing the fan-out complexity, since this inject doesn't bring new import.